File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//Graham Stodolski CS50

int main(void) {

    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++) {  //cycles through characters in text
        if (('A' <= text[i] && text[i] <= 'Z') || ('a' <= text[i] && text[i] <= 'z')) {  //checks if current character is a letter
            letters++;
        }
        else if (text[i] == ' ') {  //checks for spaces
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?') {  //checks for punctuation
            sentences++;
        }
    }

    float index = 0.0588 * (letters * 100)/(words) - 0.296 * (sentences * 100)/(words) - 15.8;  //Coleman-Liau index
    index = round(index);
    if (index < 1) {
        printf("Before Grade 1\n");
    }
    else if (index >= 16) {
        printf("Grade 16+\n");
    }
    else {
        printf("Grade %.0f\n", index);
    }
}

File: readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

for i in range(len(text)):  # cycles through characters in text
    if ("A" <= text[i] and text[i] <= "Z") or ("a" <= text[i] and text[i] <= "z"):  # checks if current character is a letter
        letters += 1
    elif text[i] == " ":  # checks for spaces to count words
        words += 1
    elif text[i] == "." or text[i] == "!" or text[i] == "?":  # checks for punctuation to count sentences
        sentences += 1

index = 0.0588 * (letters * 100) / (words) - 0.296 * (sentences * 100) / (words) - 15.8  # Coleman-Liau index
index = round(index)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(index))